### DISCUSSION TITLE
Implementing Dynamic and Multilingual Canonical Tags in Next.js 13 for SEO

### DISCUSSION THREAD

**Discussion by aleczratiu** [vercel/next.js#57794] [Help]
### Summary

Hi,

Looking for other approaches who already did an implementation in this way to have a multilingual canonical tag.
I tried a few approaches with generateMetadata in Nextjs 13 but I couldn't managed to implement a fully multilanguage canonical tag like in the below screen.

![unnamed](https://github.com/vercel/next.js/assets/33958032/c869bf46-ec9b-4905-9025-5695943452cb)

Does anyone already tried implementing the multilanguage canonical tag in Nextjs 13? What was the approach? Any suggestions?

**Comment by devjiwonchoi** [comment]
Check this out: [metadata.alternates](https://nextjs.org/docs/app/api-reference/functions/generate-metadata#alternates)

**Reply by aleczratiu** [reply]
Thank you @devjiwonchoi I'll try this.
